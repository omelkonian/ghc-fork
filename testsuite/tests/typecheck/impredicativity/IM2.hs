{-# LANGUAGE TypeOperators, TypeFamilies, DataKinds, TypeOperators, GADTs
           , MultiParamTypeClasses, FlexibleInstances, FlexibleContexts
           , ScopedTypeVariables, DoAndIfThenElse, DeriveGeneric
           , GeneralizedNewtypeDeriving, ConstraintKinds, DeriveAnyClass
           , StandaloneDeriving, ExplicitForAll, DefaultSignatures
           , ConstrainedClassMethods, BangPatterns, RankNTypes, UnicodeSyntax
           , ImpredicativeTypes
#-}

module Main (main) where

import GHC.Types -- import (<~)

t :: [Int] <~ [String] => Int
t = 1

main :: IO ()
main = print t
