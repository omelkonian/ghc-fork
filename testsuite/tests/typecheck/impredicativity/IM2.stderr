IM2.hs:14:6: error:
    • Couldn't match type ‘Int’ with ‘[Char]’
    • In the ambiguity check for ‘t’
      To defer the ambiguity check to use sites, enable AllowAmbiguousTypes
      In the type signature: t :: [Int] <~ [String] => Int
